src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -229,16 +235,9 @@ private static void HtmlDecode(string value, ref ValueStringBuilder output)
                             //      &#xE5;  --> same char in hex
                             // See http://www.w3.org/TR/REC-html40/charset.html#entities
 
-                            bool parsedSuccessfully;
-                            uint parsedValue;
-                            if (value[entityOffset + 1] == 'x' || value[entityOffset + 1] == 'X')
-                            {
-                                parsedSuccessfully = uint.TryParse(value.AsSpan(entityOffset + 2, entityLength - 2), NumberStyles.AllowHexSpecifier, CultureInfo.InvariantCulture, out parsedValue);
-                            }
-                            else
-                            {
-                                parsedSuccessfully = uint.TryParse(value.AsSpan(entityOffset + 1, entityLength - 1), NumberStyles.Integer, CultureInfo.InvariantCulture, out parsedValue);
-                            }
+                            bool parsedSuccessfully = value[entityOffset + 1] == 'x' || value[entityOffset + 1] == 'X'
```

Comment:
This is difficult to parse and read. Can we cache some of these in a local? Like `value[entityOffset + 1]`?